apps/docusaurus/docs/move/move-on-aptos/move-security-guidelines.md

```diff
@@ -589,3 +647,104 @@ public fun main(user: &signer) {
 ### Smart contract publishing key management
 
 Using the same account for testnet and mainnet poses a security risk, as testnet private keys, often stored in less secure environments (ex. laptops), can be more easily exposed or leaked. An attacker that can obtain the private key for the testnet smart contract would be able to upgrade the mainnet one.
+
+## Randomness
+
+---
+
+### Randomness - test-and-abort
+
+If a `public` function directly or indirectly invokes the randomness API, a malicious user can abuse the composability of this function and abort the transaction if the result is not as desired. This allows the user to keep trying until they achieve a beneficial outcome, undermining the randomness.
+
+#### Example Vulnerable code
+
+```move
+module user::lottery {
+    fun mint_to_user(user: &signer){
+        move_to(address_of(user),WIN{});
+    }
+
+    #[randomness]
+    public entry fun play(user: &signer) {
+        let random_value = aptos_framework::randomness::u64_range(0, 100);
+        if (random_value == 42) {
+            mint_to_user(user);
+        }
+    }
```

Comment:
Can we stay concrete and the same for these examples?

Here we use WIN, and below we use Object_for_the_winner
```suggestion
```move
module user::lottery {
    struct Winner {}

    fun mint_to_user(user: &signer) {
        move_to(address_of(user), Winner{});
    }

    #[randomness]
    public entry fun play(user: &signer) {
        let random_value = aptos_framework::randomness::u64_range(0, 100);
        if (random_value == 42) {
            mint_to_user(user);
        }
    }
```